src/main/java/hudson/plugins/deploy/CargoContainerAdapter.java

```diff
@@ -93,37 +94,51 @@ protected WAR createWAR(File deployableFile) {
         return new WAR(deployableFile.getAbsolutePath());
     }
 
+    /**
+     * Expands an encoded environment variable. Ex. if HOME=/user/alex, expands '${HOME}' to '/user/alex'
+     *
+     * @param envVars the environment variables of the build
+     * @param resolver unused
+     * @param variable the variable to expand
+     * @return the value of the expanded variable
+     */
     protected String expandVariable(EnvVars envVars, VariableResolver<String> resolver, String variable) {
-        String temp = envVars.expand(variable);
         return Util.replaceMacro(envVars.expand(variable), resolver);
     }
 
-
     /**
      * Creates a Deployable object EAR from the given file object.
      *
      * @param deployableFile The deployable file to create the Deployable from.
-     * @return A Deployable object.
+     * @return A deployable object.
      */
     protected EAR createEAR(File deployableFile) {
         return new EAR(deployableFile.getAbsolutePath());
     }
 
+    @Override
     public boolean redeploy(FilePath war, final String contextPath, final AbstractBuild<?, ?> build, Launcher launcher, final BuildListener listener) throws IOException, InterruptedException {
-        return war.act(new FileCallable<Boolean>() {
-            public Boolean invoke(File f, VirtualChannel channel) throws IOException {
-                if (!f.exists()) {
-                    listener.error(Messages.DeployPublisher_NoSuchFile(f));
-                    return true;
-                }
+        return redeploy(war, contextPath, build, launcher, (TaskListener)listener);
+    }
+
+    public boolean redeploy(FilePath war, final String contextPath, final Run<?, ?> run, final Launcher launcher, final TaskListener listener) throws IOException, InterruptedException {
+        return war.act(new MasterToSlaveFileCallable<Boolean>() {
+            @Override
+            public Boolean invoke(File f, VirtualChannel channel) throws IOException, InterruptedException {
+
                 ClassLoader cl = getClass().getClassLoader();
                 final ConfigurationFactory configFactory = new DefaultConfigurationFactory(cl);
                 final ContainerFactory containerFactory = new DefaultContainerFactory(cl);
                 final DeployerFactory deployerFactory = new DefaultDeployerFactory(cl);
 
                 try {
-                    final EnvVars envVars = build.getEnvironment(listener);
-                    final VariableResolver<String> resolver = build.getBuildVariableResolver();
+                    final EnvVars envVars;
+                    if (run instanceof AbstractBuild) {
+                        envVars = run.getEnvironment(listener);
+                    } else {
+                        envVars = new EnvVars(); // empty for Workflow
+                    }
+                    final VariableResolver<String> resolver = new VariableResolver.ByMap<String>(envVars);
```

Comment:
Nope, you need to use `build.getBuildVariableResolver()`. It is _not_ the same as the build’s “environment”.